Lib/textwrap.py

```diff
@@ -14,6 +14,40 @@
 # some Unicode spaces (like \u00a0) are non-breaking whitespaces.
 _whitespace = '\t\n\x0b\x0c\r '
 
+try:
+    from unicodedata import east_asian_width
+
+    def _len(text):
+        """Return the display length of the text in columns, according to
+        unicodedata.east_asian_width only.
+        """
+        return sum(2 if east_asian_width(char) in {'F', 'W'} else 1
+                 for char in text)
+
+    def _slice(text, index):
+        """Return the two slices of text cut to index.
+        """
+        width = 0
+        pos = 0
+        for char in text:
+            width += 2 if east_asian_width(char) in {'F', 'W'} else 1
+            if width > index:
+                break
+            pos += 1
```

Comment:
Why note use enumerate(), it's less readable (less pythonic)

Reply:
Because it does not works with enumerate as the last incrementation were not done. I do not remember which case exactly but if you run the unit test you'll spot it easily, it was failing, I'll do if needed but can't right now.

Reply:
Interested in that, the code was strongly tested for txt2tags and don't catch this problem.

Reply:
Your initial implementation was working thanks to your `if cjk_len(text) <= index: return text, ''` fixing the special case explicitly, I may have tried to avoid it.

Reply:
"Explicit is better than implicit." but the more important is that both solutions are correct.